protractor_demo/pages/search.ts

```diff
@@ -1,22 +1,26 @@
 import { Navigation } from '../widgets/navigation';
-import { browser, ExpectedConditions, protractor } from 'protractor';
+import { $, browser, ElementFinder, ExpectedConditions, protractor } from 'protractor';
 
 export class Search {
+    private readonly navigation: Navigation;
+    private readonly input: ElementFinder;
 
-    private readonly navigationMenu = new Navigation();
-    private readonly searchInput = browser.$('[name="q"]');
+    constructor() {
+        this.navigation = new Navigation();
+        this.input = $('[name="q"]');
+    }
 
-    async open() {
-        await browser.get('http://google.com/ncr');
+    public async open(url: string = 'http://google.com/ncr'): Promise<void> {
```

Comment:
url for google search page is static - you dont need to pass anything in method which should open page (url especially)